runner.py

```diff
@@ -1,12 +1,39 @@
-"""Script that is able to start Uvicorn-based REST API service."""
+"""Main app entrypoint. Starts Uvicorn-based REST API service."""
 
+import logging
 import os
+from pathlib import Path
 
 import uvicorn
 
-if __name__ == "__main__":
-    # TODO syedriko consolidate this env var into a central location as per OLS-345.
-    # disable Gradio analytics, which calls home to https://api.gradio.app
+from ols.utils import config
+from ols.utils.logging import configure_logging
+
+
+def configure_3rd_party_envs() -> None:
+    """Configure environment variables for our dependencies."""
+    # disables Gradio analytics, which calls home to https://api.gradio.app
     os.environ["GRADIO_ANALYTICS_ENABLED"] = "false"
 
+    # configures envs for HuggingFace library
+    os.environ["TRANSFORMERS_OFFLINE"] = "1"
+    os.environ["HF_HOME"] = (
+        config.ols_config.reference_content.embeddings_model_path or ""
+    )
+
+
+if __name__ == "__main__":
+
+    cfg_file = os.environ.get("OLS_CONFIG_FILE", "olsconfig.yaml")
+    config.init_config(cfg_file)
+
+    configure_logging(config.ols_config.logging_config)
+    logger = logging.getLogger(__name__)
+    logger.info(f"Config loaded from {Path(cfg_file).resolve()}")
+
+    configure_3rd_party_envs()
```

Comment:
It is safer to set these as soon as possible, before we do anything else. For example, Gradio looks at GRADIO_ANALYTICS_ENABLED at package initialization time:
https://github.com/gradio-app/gradio/blob/24626ea362a35606d344f341a308556c52b34cc0/gradio/strings.py#L39